server/sttp-stub-server/src/main/scala/sttp/tapir/server/stub/SttpRequestBody.scala

```diff
@@ -66,4 +77,54 @@ class SttpRequestBody[F[_]](implicit ME: MonadError[F]) extends RequestBody[F, A
     transfer()
     os.toByteArray
   }
+
+  private def extractMultipartParts(parts: Seq[Part[client3.RequestBody[_]]], bodyType: RawBodyType.MultipartBody): List[Part[Any]] = {
+    parts.flatMap { part =>
+      bodyType.partType(part.name).flatMap { partType =>
+        val body = extractPartBody(part, partType)
+        Some(Part(
+          name = part.name,
+          body = body,
+          contentType = part.contentType.flatMap(ct => MediaType.parse(ct).toOption),
+          fileName = part.fileName
+        ))
+      }
+    }.toList
+  }
+
+  private def extractPartBody[B](part: Part[client3.RequestBody[_]], bodyType: RawBodyType[B]): Any = {
+    part.body match {
+      case ByteArrayBody(b, _) =>
+        bodyType match {
+          case RawBodyType.StringBody(charset)  => b
+          case RawBodyType.ByteArrayBody        => b
+          case RawBodyType.ByteBufferBody       => ByteBuffer.wrap(b)
+          case RawBodyType.InputStreamBody      => new ByteArrayInputStream(b)
+          case RawBodyType.InputStreamRangeBody => InputStreamRange(() => new ByteArrayInputStream(b))
+          case RawBodyType.FileBody             => throw new IllegalArgumentException("ByteArray body provided while endpoint accepts FileBody")
+          case _: RawBodyType.MultipartBody     => ME.error(new IllegalArgumentException)
```

Comment:
hm that's not how monads are used in Scala (especially when returning `Any`), the errors won't get propagated back. Probably throwing exceptions is the easier and more readable thing to do here, sorry for the suggestion